kotlin/samples/dungeon/app/src/main/java/com/squareup/sample/dungeon/GameLayoutRunner.kt

```diff
@@ -54,13 +52,7 @@ class GameLayoutRunner(
   }
 
   override fun showRendering(rendering: GameRendering) {
-    if (!::boardView.isInitialized) {
-      boardView = viewRegistry.buildView(rendering.board, boardContainer)
-      boardContainer.addView(boardView)
-    } else {
-      boardView.showRendering(rendering.board)
-    }
-
+    boardView = viewRegistry.updateView(boardView, rendering.board)
```

Comment:
For this case, we could make a property delegate like:

```kotlin
private val boardView by viewRegistry.bindStub<GameBoard>(R.id.board_stub)
// Or
private val boardView = viewRegistry.bindStub<GameBoard>(R.id board_stub)

…

boardView.update(rendering.board)
// Or even
boardView = rendering.board
```

Reply:
But we still need an update method for the modal case below. Seems easy to forget to reassign from the return value - instead the update can just take a `KMutableProperty` and then you just pass in the property with `::baseView`.

Reply:
But we still need an update method for the modal case below. Seems easy to forget to reassign from the return value - instead the update can just take a `KMutableProperty` and then you just pass in the property with `::baseView`.

Reply:
Finally starting to grok this, maybe I do like it better than `WorkflowViewStub`. Also, holy shit, `KMutableProperty` is our ticket to Swift inout parameters, no?

Reply:
I'll finish `WorkflowViewStub` before exploring this.

Reply:
Only a tiny aspect of them, and the least interesting part. An early implementation of what is now called `Mutator` used them, but that wasn't very interesting because it wasn't public-facing code. 

The real win with `inout` is that you can pass structs that have `var` properties, assign to those properties like they were vars on a mutable reference type, and automatically see those changes in the caller, all without losing any purity.